cmd/grype-db/cli/commands/cache_backup.go

```diff
@@ -138,13 +141,25 @@ func archiveProvider(root string, name string, writer *tar.Writer) error {
 
 	return filepath.Walk(name,
 		func(path string, info os.FileInfo, err error) error {
-			if err != nil {
-				return err
-			}
-
 			if info.IsDir() {
 				return nil
 			}
+			if strings.HasSuffix(path, "metadata.json") && cfg.Results.ResultsOnly {
+				// Mark metadata stale
+				state, err := provider.ReadState(path)
+				if err != nil {
+					return err
+				}
+
+				state.Stale = true
+				err = provider.WriteState(path, state)
+				if err != nil {
+					return err
+				}
+			}
+			if cfg.Results.ResultsOnly && strings.Contains(path, "input") {
```

Comment:
this will exclude any path that has `input` in it, and though this works today, we are really only intending to exclude `./input` relative to the root of the workspace.